server/scripts/20230803172011-migrate-emoji-in-document-title.ts

```diff
@@ -0,0 +1,59 @@
+import "./bootstrap";
+import { Transaction } from "sequelize";
+import parseTitle from "@shared/utils/parseTitle";
+import { sequelize } from "@server/database/sequelize";
+import { Document } from "@server/models";
+
+const limit = 100;
+let page = parseInt(process.argv[2], 10);
+page = Number.isNaN(page) ? 0 : page;
+
+export default async function main(exit = false) {
+  const work = async (page: number): Promise<void> => {
+    console.log(`Backfill document emoji from title… page ${page}`);
+    await sequelize.transaction(async (transaction) => {
+      const documents = await Document.unscoped().findAll({
+        attributes: {
+          exclude: ["state"],
+        },
+        limit,
+        offset: page * limit,
+        order: [["createdAt", "ASC"]],
+        paranoid: false,
+        lock: Transaction.LOCK.UPDATE,
+        transaction,
+      });
+
+      for (const document of documents) {
+        try {
+          const { emoji, strippedTitle } = parseTitle(document.title);
+          if (emoji) {
+            document.emoji = emoji;
+            document.title = strippedTitle;
+
+            await document.save({
+              silent: true,
+              transaction,
+            });
+          }
+        } catch (err) {
+          console.error(`Failed at ${document.id}:`, err);
+          continue;
+        }
+      }
+
+      return documents.length === limit ? work(page + 1) : undefined;
```

Comment:
Because `work` is returned here recursively the transaction is never closed, we need to make sure that the transaction is committed before the next iteration of `work` is called

Reply:
Good catch! Better to run it iteratively then I guess, page by page?

Reply:
Just move the return statement outside of the sequelize transaction and it will solve it